hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -167,6 +167,11 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute(HoodieData<HoodieRec
 
   @Override
   public HoodieWriteMetadata<HoodieData<WriteStatus>> execute(HoodieData<HoodieRecord<T>> inputRecords, Option<HoodieTimer> sourceReadAndIndexTimer) {
+    int coalesceParallelism = config.getMetadataConfig().getWriteStatusCoalesceParallelism();
```

Comment:
we can just name this `parallelism`. 

and 
```
if (table.isMetadataTable() && parallelism > 0) {
  inputRecords = inputRecords.coalesce(parallelism); 
} 
```

Reply:
actually, lets not add this fix here. 
lets move this to HoodieBackedTableMetadatawriter.commitInternal() just before calling writeclient.upsertPreppred(...)